File: Java/13-RomanToInteger.java
class Solution {
    public int romanToInt(String s) {
    	
    	int num = 0;
    	
    	// take care of subtraction
    	boolean skip_next = false;
    	
    	for (int i = 0; i < s.length(); i++){
    		
    	    char c = s.charAt(i);
    	    char next_char = 'a';
    	    
    	    if(i < (s.length() - 1)) {
    	    	next_char = s.charAt(i + 1);
    	    }
    	    
    	    if(!skip_next) {
    	    	switch(c) {
    	    	case 'I':
    	    		if(next_char == 'V') {
                        skip_next = true;
                        num += 4;
    	    		}
                        
    	    		else if(next_char == 'X') {
                        skip_next = true;
                        num += 9;
    	    		}
                    
                    else {
                        num += 1;
                    }
    	    		continue;
    	    		
    	    	case 'V':
    	    		num += 5;
    	    		continue;
    	    		
    	    	case 'X':
    	    		if(next_char == 'L') {
                        skip_next = true;
                        num += 40;
    	    		}
                        
    	    		else if(next_char == 'C') {
                        skip_next = true;
                        num += 90;
    	    		}
                    
                    else {
                        num += 10;
                    }
    	    		continue;
    	    		
    	    	case 'L':
    	    		num += 50;
    	    		continue;
    	    		
    	    	case 'C':
    	    		if(next_char == 'D') {
                        skip_next = true;
                        num += 400;
    	    		}
                        
    	    		else if(next_char == 'M') {
                        skip_next = true;
                        num += 900;
    	    		}
                    
                    else {
                        num += 100;
                    }
    	    		continue;
    	    		
    	    	case 'D':
    	    		num += 500;
    	    		continue;
    	    		
    	    	case 'M':
    	    		num += 1000;
    	    		continue;
    	    	}
    	    	
    	    }
    	    
    	    else {
    	    	skip_next = false;
    	    }
    	}
    	
    	return num;
        
    }
}

File: Python/13-RomanToInteger.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        num = 0
        
        # for subtraction cases
        skip_next = False
        
        for i, char in enumerate(s):
            
            if i < len(s) - 1:
                next_char = s[i + 1]
            else:
                next_char = ""
            
            if not skip_next:
                if char == "I":
                    if next_char == "V":
                        skip_next = True
                        num += 4
                        
                    elif next_char == "X":
                        skip_next = True
                        num += 9
                    
                    else:
                        num += 1
                    
                elif char == "V":
                    num += 5
                    
                elif char == "X":
                    if next_char == "L":
                        skip_next = True
                        num += 40
                        
                    elif next_char == "C":
                        skip_next = True
                        num += 90
                    
                    else:
                        num += 10
                    
                elif char == "L":
                    num += 50
                    
                elif char == "C":
                    if next_char == "D":
                        skip_next = True
                        num += 400
                        
                    elif next_char == "M":
                        skip_next = True
                        num += 900
                    
                    else:
                        num += 100
                    
                elif char == "D":
                    num += 500
                    
                elif char == "M":
                    num += 1000
                    
            else:
                skip_next = False
            
        return num